### PR TITLE
Improvements to song sorting and updated navigation bar button names

### PR DESCRIPTION
**Pull request by SamuelPalma**
* The navigation bar is updated to show the following action for pressing the search and sort buttons

* Improvement of the sorting code and addition of sorting by year, genre, album and charter.

* The code that performs song sorting (SongSorting.cs) is separated from the class that performs song selection (SongSelection.cs).

### CODE REVIEW

**Comment by SamuelPalma** [comment]
YARG_20230602_085354

File: Assets/Script/UI/MusicLibrary/SongSelection.cs [hunk 1]
```diff
@@ -537,43 +562,31 @@ private enum SortCriteria {
...
-			bool isLast = indexOfActualLetter == (songsFirstLetter.Count - 1);
+		private int GetSkip(){
+			return Mathf.Max(1, _songs.Count - SongContainer.Songs.Count);
+		}
 
-			if (isLast) {
-				var firstCharacterInList = char.ToString(songsFirstLetter[0]);
-				return firstCharacterInList;
-			}
+		private string getNextSortCriteriaButtonName() {
```
Comment (EliteAsian123) [inline Assets/Script/UI/MusicLibrary/SongSelection.cs:584]: Follow naming conventions (should be `GetNextSortCriteriaButtonName`)

File: Assets/Script/UI/MusicLibrary/SongSelection.cs [hunk 2]
```diff
@@ -115,7 +111,28 @@ private enum SortCriteria {
...
+
+			searchBoxShouldBeEnabled = true;
+
+			// Play preview on enter for selected song
+			if (_songs[SelectedIndex] is SongViewType song) {
+				_previewContext.PlayPreview(song.SongEntry).Forget();
+			}
+		}
+
+		private NavigationScheme getNavigationScheme(){
```
Comment (EliteAsian123) [inline Assets/Script/UI/MusicLibrary/SongSelection.cs:134]: Follow naming conventions (should be `GetNavigationScheme`)

File: Assets/Script/UI/MusicLibrary/SongSelection.cs [hunk 3]
```diff
@@ -172,36 +173,66 @@ private enum SortCriteria {
...
+		public SongSorting.SortCriteria getNextSortCriteria() {
+			return SongSorting.Instance.GetNextSortCriteria(_sortCriteria);
+		}
+
+		private void UpdateNextSortCriteria(){
+			_nextSortCriteria = getNextSortCriteriaButtonName();
+		}
+
+		private void UpdateNavigationScheme(){
+			Navigator.Instance.PopAllSchemes();
```
Comment (EliteAsian123) [inline Assets/Script/UI/MusicLibrary/SongSelection.cs:197]: Should probably just be `Navigator.Instance.PopScheme();`. `PopAllSchemes` gets rid of other potential schemes that are on the scheme stack, which could result in an unnavigable menu in the future.

**Review by EliteAsian123 — CHANGES_REQUESTED** [review]

**Review by EliteAsian123 — APPROVED** [review]

**Pull request merged**